### ISSUE TITLE
[QA] OCM - images (& thumbnails) are completely broken with OCM

### ISSUE THREAD

**Issue by jnweiger** [owncloud/ocis#10272]
## Describe the bug
A user who received a shared folder via federation cannot view images. The image viewer hangs

## Steps to reproduce
Setup two federated machines with e.g.

env OCIS_DNSNAME=dog-DATE OCIS_VERSION=daily ./deploy_ocis_bare_metal.sh
env OCIS_DNSNAME=cat-DATE OCIS_VERSION=daily ./deploy_ocis_bare_metal.sh
ocm_trust.sh dog-20241015.jw-qa.owncloud.works cat-20241015.jw-qa.owncloud.works

* admin adds the instances to each other's ocmproviders.json .
* user alice at cat visits the sciencemesh app and create an invite token for user bob at dog.
* user bob at dog enters the token, and shares a folder smallpics (continaing some png and jpg images) with alice ad cat
* user alice clicks Shares in the left sidebar, and clicks smallpics ( see #10271 )
* user alice sees no thumbnails in the folder view: BAD

![image](https://github.com/user-attachments/assets/ccb12963-b10f-4848-90bc-d219b476be33)
----

* user alice clicks on one of the images. It does not disaplay in the viewer: BAD

![image](https://github.com/user-attachments/assets/77e13063-483c-4af1-9eac-f73c5a7d6f05)
----

* user alice tries to download an image. A login prompt appears: BAD

![image](https://github.com/user-attachments/assets/b115edb1-d184-4e41-8320-33b28c395a16)
----

* after entering the correct credentials there, the download works. OKish
* Image viewer still does not. BAD

* txt and md files work perfectly fine.

## Expected behavior
* thumbnails appear for federated folders too,
* images can be viewed and downloaded.

Seen with both ocis-6.5.0 and daily 2024-10-09

**Comment by JammingBen** [comment]
There are 2 underlying issues:

* When sharing a single image file, the `file.mimetype` property of the `driveItem` (returned by the server when querying `graph/v1beta1/me/drive/sharedWithMe`) is `application/octet-stream`. So Web is not recognizing the file as an image.

* When sharing a folder that includes an image, the `GET` request to the image fails with a `404`. It looks correct to me though: `https://localhost:9200/remote.php/dav/spaces/{remoteItemId}/image.jpg`

IMO this is a p2, images are completely broken in OCM. This does not only affect thumbnails, but also viewing images via the preview app. @micbar agree?

**Comment by micbar** [comment]
# Scope

- Previews and Thumbnails can be broken -> Known Issue
- Downloads and Archiver need to work.

**Comment by ScharfViktor** [comment]
- [ ] TODO: create apiTest for downloading file/folder/tar
- [ ] TODO: create e2e test for Previews and Thumbnails after https://github.com/owncloud/web/pull/11693 will be done

**Comment by micbar** [comment]
@butonic @kwdrago @kobergj @rhafer 
After talking today, we came to the conclusion that this is fixable in the scope of the 7.0.0 release. OCM has a storageprovider implemented with a fixed providerID, so the thumbnail and archiver services should be able to use the default implementation by doing an `InitiateFileDownload`request to the gateway. We have a bug in the implementation maybe, but no "missing" piece.

**Comment by rhafer** [comment]
> After talking today, we came to the conclusion that this is fixable in the scope of the 7.0.0 release.

If you think so ...

```
{"level":"error","service":"thumbnails","time":"2024-10-30T15:37:01Z","line":"github.com/owncloud/ocis/v2/services/thumbnails/pkg/service/grpc/v0/service.go:311","message":"resource info is missing checksum"}
```

The first issue seems to be that the ocm storage provider does not return checksums in its StatResponse. Still trying to figure out why exactly that is...

**Comment by rhafer** [comment]
Fix was merged.

**Comment by jnweiger** [comment]
Confirmed fixed in 7.0.0-rc.3 + web 11.0.4
